This patch is based on upstream commit
ae268a2b1aaa14e16ffc47b3cd5ad74a26e39838 and fixes JB#32512.

From ae268a2b1aaa14e16ffc47b3cd5ad74a26e39838 Mon Sep 17 00:00:00 2001
From: Umut Tezduyar Lindskog <umut@tezduyar.com>
Date: Tue, 2 Sep 2014 09:02:31 +0200
Subject: [PATCH] enable build support without systemd compatibility libraries

systemd 209 merged all the libraries to libsystemd. Old
libraries can still be enabled with --enable-compat-libs
switch in systemd but this increases the binary size.

Implement a fallback library check in case compat libraries
dont exist.

[Fixed underquoting; switched priority so we try libsystemd first -smcv]
Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 configure.ac | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5412db2..b7b91be 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1169,10 +1169,13 @@ dnl systemd detection
 if test x$enable_systemd = xno ; then
     have_systemd=no;
 else
-    PKG_CHECK_MODULES(SYSTEMD,
-        [libsystemd-login >= 32, libsystemd-daemon >= 32],
-        have_systemd=yes,
-        have_systemd=no)
+    PKG_CHECK_MODULES([SYSTEMD],
+        [libsystemd >= 209],
+        [have_systemd=yes],
+        [PKG_CHECK_MODULES([SYSTEMD],
+            [libsystemd-login >= 32, libsystemd-daemon >= 32, libsystemd-journal >= 32],
+            [have_systemd=yes],
+            [have_systemd=no])])
 fi
 
 if test x$have_systemd = xyes; then
-- 
2.1.4

